File: src/main/java/com/tencao/saoui/api/elements/animator/internal/InvocationWrapper.java
package com.tencao.saoui.api.elements.animator.internal;

/**
 * This is a copy from the library LibrarianLib
 * This code is covered under GNU Lesser General Public License v3.0
 */

import java.lang.invoke.MethodHandle;

/**
 * @author WireSegal
 * Created at 7:38 PM on 10/22/16.
 */
/*package-private*/ class InvocationWrapper {
    private MethodHandle handle;

    /*package-private*/ InvocationWrapper(MethodHandle handle) {
        this.handle = handle;
    }

    /*package-private*/ Object invoke() throws Throwable {
        return handle.invokeExact();
    }

    /*package-private*/ Object invoke(Object obj) throws Throwable {
        return handle.invokeExact(obj);
    }

    /*package-private*/ Object invoke(Object obj, Object second) throws Throwable {
        return handle.invokeExact(obj, second);
    }

    /*package-private*/ Object invokeArity(Object[] args) throws Throwable {
        return handle.invokeExact(args);
    }
}

File: src/main/java/com/saomc/saoui/api/elements/old/InvocationWrapper.kt
package com.saomc.saoui.api.elements.old

import java.lang.invoke.MethodHandle

/**
 * @author WireSegal
 * Created at 7:38 PM on 10/22/16.
 */
/*package-private*/
@Deprecated("Update with LibrarianLib")
internal class InvocationWrapper     /*package-private*/(private val handle: MethodHandle) {
    /*package-private*/
    @Throws(Throwable::class)
    operator fun invoke(): Any {
        return handle.invokeExact()
    }

    /*package-private*/
    @Throws(Throwable::class)
    operator fun invoke(obj: Any?): Any {
        return handle.invokeExact(obj)
    }

    /*package-private*/
    @Throws(Throwable::class)
    operator fun invoke(obj: Any?, second: Any?): Any {
        return handle.invokeExact(obj, second)
    }

    /*package-private*/
    @Throws(Throwable::class)
    fun invokeArity(args: Array<Any?>): Any {
        return handle.invokeExact(*args)
    }
}
